beacon-chain/state/state-native/getters_withdrawal.go

```diff
@@ -132,9 +134,16 @@ func (b *BeaconState) ExpectedWithdrawals() ([]*enginev1.Withdrawal, uint64, err
 				return nil, 0, fmt.Errorf("could not retrieve balance at index %d: %w", w.Index, err)
 			}
 			hasSufficientEffectiveBalance := v.EffectiveBalance() >= params.BeaconConfig().MinActivationBalance
-			hasExcessBalance := vBal > params.BeaconConfig().MinActivationBalance
+			var totalWithdrawn uint64
+			for _, wi := range withdrawals {
+				if wi.ValidatorIndex == w.Index {
+					totalWithdrawn += wi.Amount
+				}
+			}
+			balance := vBal - totalWithdrawn
```

Comment:
What is preventing an underflow here?

Reply:
the hasExcessBalance check with the min check on amount will always make sure it has at least the min activation balance.

Reply:
But you do that after the subtraction?